Ghidra/Features/Base/src/main/java/ghidra/app/plugin/core/datamgr/archive/FileArchive.java

```diff
@@ -283,7 +283,7 @@ public void saveAs(Component component) throws IOException {
 			return;
 		}
 
-		if (saveAsFile.equals(getFile())) {
+		if (saveAsFile.getName().equals(getFile().getName())) {
```

Comment:
this needs to change to compare the full file path, not just the name.

Suggest ```saveAsFile.equals(archiveFile.getFile(false))```